front/app/utils/i18n.ts

```diff
@@ -66,6 +66,12 @@ export function getLocalized(
 
     if (isObject(multiloc) && !isEmpty(multiloc)) {
       const multilocLocales = keys(multiloc) as Locale[];
+
+      const sameLanguageLocale = findSimilarLocale(locale, multilocLocales);
```

Comment:
The method `getLocalized` became so complicated (I mean not just after adding your code, but in general). If you investigated and completely figured it out, it would be nice to split it into multiple methods with clear names. If it's not easy, don't bother.

Reply:
Actually, I really don't get it and I'm starting to think something might be wrong with the current solution (when is the value for exactly the requested locale tried first?). I asked Luuc if he could verify.

Reply:
I think there's now a potential bug in here: if your locale is `'en'`, and your multiloc has `'en'` and `'en-CA'` for example, there's no guarantee that is actually returns the correct locale (it could return either `'en'` or `'en-CA'`, depending on what comes first in the `multilocLocales` array). So perhaps you should add some logic before `findSimilarLocale`, e.g.

```ts
if (isObject(multiloc) && !isEmpty(multiloc)) {
  if (!!multiloc[locale]) return truncate(multiloc[locale], maxLength) // this line

  const multilocLocales = keys(multiloc) as Locale[];

  const sameLanguageLocale = findSimilarLocale(locale, multilocLocales);
  if (!isNilOrError(sameLanguageLocale) && !!multiloc[sameLanguageLocale]) {
    return truncate(multiloc[sameLanguageLocale], maxLength);
  }
```

Also, I would rename `sameLanguageLocale` to `similarLanguageLocale`, especially if you add the check I add above where it's ruled out that it's the same language. I also wouldn't use `isNilOrError` when it can't be an error, there's a `isNil` utility in the same utils file.

I also agree with Alex that breaking this stuff up in sub-functions would be nice since it's becoming quite a complex beast, but if you don't feel comfortable doing that in TS I get that.

Finally, I am actually making some changes to this file on the topic filters branch. It also contains unit tests for this file, which might be useful to make sure you're not breaking anything. Maybe it makes sense to wait with merging this until my branch with the unit tests is merged? I can help you get rid of any merge conflicts, and also help with refactoring this file a bit. But if this is an urgently needed fix we can think of another solution